### ISSUE TITLE
fx128: Popups no longer auto-hide in dialogs on macOS

### ISSUE THREAD

**Issue by AbeJellinek** [zotero/zotero#4891]
May be related to non-native menupopup (@windingwind)? I'm not sure.

1. Right-click an item -> Export Item...
2. Click the Format menulist
3. Click somewhere else, or Cmd-Tab

Popup should disappear, but it stays open until you press Escape. Other non-native popups work fine - this seems to only affect popups in dialog windows.

**Comment by windingwind** [comment]
No idea why's that... Sorry I did some tries but forgot to reply. Should not be related to our custom element mixins.

**Comment by mrtcode** [comment]
After testing the fx128 branch and then switching back to the main branch using the same profile, I found that it had set `ui.popup.disable_autohide` and `devtools.popup.disable_autohide` to `true`.

It also changed `extensions.zotero.openReaderInNewWindow` to `true`.

So this may be related. Also who knows what other properties might have been altered.

**Comment by dstillman** [comment]
I don't think this is related to the `disable_autohide` prefs — those haven't been changed for me, and that setting would affect all popups.

**Comment by abaevbog** [comment]
Are we sure this is specific to `fx128`? I am observing the same behavior on both `fx128` branch and in the latest beta where the first time a dialog is opened, its popup will auto-hide if you click elsewhere but on subsequent times, it won't.

This is what I see on both fx128 and the current beta:
- start Zotero
- Right-click on an item -> Export Item -> Format (popup appears)
- click on the body of the dialog
- it does go away
- click cancel, dialog closes
- Right-click on an item -> Export Item -> Format (popup appears)
- click on the body of the dialog
- popup does not go away

Another observation I made: 
- start Zotero
- focus some other window
- return focus to Zotero window
- Right-click on an item -> Export Item -> Format (popup appears)
- click on the body of the dialog
- even this first time, it does not go away

Maybe it has something to do with focus handling on the main window? Perhaps it's related to this other dialog focus-related issue: https://github.com/zotero/zotero/issues/4819?

**Comment by abaevbog** [comment]
Oh, one more thing: `newCollectionDialog` can be opened by pressing the `toolbarbutton` above collectionTree or by pressing `+` in itemPane "Libraries and Collections" section and selecting "New Collection" in the popup. I noticed that in the first case, the popup with collections will auto-hide as expected (e.g. if I click on the dialog). But if I open `newCollectionDialog` through "New Collection" context menu, then menulist popup will not auto-hide after the first time as described above.

That made me think that maybe there is some kind of conflict or race condition happening when multiple popups are involved? It does look like it because if I add `Zotero.Promise.delay();` right before opening the "Export Items" popup here https://github.com/zotero/zotero/blob/85deee9631a93cab7d4a87461214da8d24a58337/chrome/content/zotero/fileInterface.js#L109-L110
this seems to fix the auto-hide for me. Same for the `newCollectionDialog`.
